### PR TITLE
Implement Azure Application Gateway create template deployment

### PR DESCRIPTION
**Pull request by milismsft**
Implement Azure Application Gateway create template deployment; unit test cases to test the implementation included as well.

Add subnet as part of the description for Spinnaker load balancer.

@JargoonPard @scotmoor PTAL

### CODE REVIEW

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/appgateway/model/AzureAppGatewayDescription.groovy [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  List<AzureAppGatewayHealthcheckProbe> probes = []
+  List<AzureAppGatewayRule> rules = []
+  // TODO: remove hardcoded sku, tier and capacity
+  //   We will come back and revisit this if we need to support different selection (we will also need deck changes to reflect that)
+  String sku = "Standard_Small"
+  String tier = "Standard"
+  long capacity = 2
+
+  static class AzureAppGatewayHealthcheckProbe {
+    enum AzureLoadBalancerProbesType {
```
Comment (JargoonPard) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/appgateway/model/AzureAppGatewayDescription.groovy]: This should just be Protocol not AzureAppGatewayHealthcheckProbe and then can be reused in multiple places.

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/appgateway/model/AzureAppGatewayDescription.groovy [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    AzureLoadBalancerProbesType protocol = "HTTP"
+    String host = "localhost"
+    String path
+    long interval = 120
+    long timeout = 30
+    long unhealthyThreshold = 8
+  }
+
+  static class AzureAppGatewayRule {
+    enum AzureLoadBalancingRulesType {
```
Comment (JargoonPard) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/appgateway/model/AzureAppGatewayDescription.groovy]: Same comment as above on AzureAppGatewayHealthcheckProbe

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/appgateway/model/AzureAppGatewayDescription.groovy [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  long capacity = 2
+
+  static class AzureAppGatewayHealthcheckProbe {
+    enum AzureLoadBalancerProbesType {
+      HTTP
+    }
+
+    String name
+    AzureLoadBalancerProbesType protocol = "HTTP"
+    String host = "localhost"
```
Comment (JargoonPard) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/appgateway/model/AzureAppGatewayDescription.groovy]: Did you confirm that localhost will resolve correctly?

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/appgateway/model/AzureAppGatewayDescription.groovy [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import com.netflix.spinnaker.clouddriver.azure.resources.common.AzureResourceOpsDescription
+
+class AzureAppGatewayDescription extends AzureResourceOpsDescription {
+  String loadBalancerName
+  String vnet
+  String subnet
+  String securityGroup
+  String dnsName
+  String cluster
+  List<String> serverGroups
```
Comment (JargoonPard) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/appgateway/model/AzureAppGatewayDescription.groovy]: vnet to serverGroups, these values aren't ever passed in to an AppGateway for creation, why are they needed?

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/loadbalancer/model/AzureLoadBalancer.groovy [hunk 5]
```diff
@@ -27,6 +27,7 @@ class AzureLoadBalancer implements LoadBalancer {
   String name
   String region
   String vnet
+  String subnet
```
Comment (JargoonPard) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/loadbalancer/model/AzureLoadBalancer.groovy]: We don't render this anywhere why was it added?

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/loadbalancer/model/AzureLoadBalancerDescription.groovy [hunk 6]
```diff
@@ -21,6 +21,7 @@ import com.netflix.spinnaker.clouddriver.azure.resources.common.AzureResourceOps
 class AzureLoadBalancerDescription extends AzureResourceOpsDescription {
   String loadBalancerName
   String vnet
+  String subnet
```
Comment (JargoonPard) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/loadbalancer/model/AzureLoadBalancerDescription.groovy]: We don't render this anywhere why was it added?

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/loadbalancer/view/AzureLoadBalancerController.groovy [hunk 7]
```diff
@@ -73,6 +73,7 @@ class AzureLoadBalancerController {
       lbDetail.createdTime = azureLoadBalancerDescription.createdTime
       lbDetail.serverGroup = azureLoadBalancerDescription.serverGroup
       lbDetail.vnet = azureLoadBalancerDescription.vnet ?: "vnet-unassigned"
+      lbDetail.subnet = azureLoadBalancerDescription.subnet ?: "subnet-unassigned"
```
Comment (JargoonPard) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/loadbalancer/view/AzureLoadBalancerController.groovy]: We don't render this anywhere why was it added?

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/loadbalancer/view/AzureLoadBalancerProvider.groovy [hunk 8]
```diff
@@ -78,6 +78,7 @@ class AzureLoadBalancerProvider implements LoadBalancerProvider<AzureLoadBalance
       name: loadBalancerDescription.loadBalancerName,
       region: loadBalancerDescription.region,
       vnet: loadBalancerDescription.vnet?: "vnet-unassigned",
+      subnet: loadBalancerDescription.subnet?: "subnet-unassigned",
```
Comment (JargoonPard) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/loadbalancer/view/AzureLoadBalancerProvider.groovy]: We don't render this anywhere why was it added?

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/ops/CreateAzureServerGroupAtomicOperation.groovy [hunk 9]
```diff
@@ -110,6 +110,7 @@ class CreateAzureServerGroupAtomicOperation implements AtomicOperation<Map> {
       lbDescription.detail = description.detail
       lbDescription.securityGroup = description.securityGroup?.name
       lbDescription.vnet = virtualNetworkName
+      lbDescription.subnet = subnetName
```
Comment (JargoonPard) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/ops/CreateAzureServerGroupAtomicOperation.groovy]: We don't render this anywhere why was it added?

**Comment by ghost** [comment]
LGTM

**Pull request closed without merging**